memory/notes/CombatManager.md

```diff
@@ -0,0 +1,101 @@
+
+- TODO: Character HP
+- TODO: Boost (AP) on ground
+- Cutscenes
+  - There is information on cutsceneRunner and encounterCutscenes, this is probably for boss fights, and we'll need this information to proceed when its a blocker.
+- Timed Attacks
+  - Theres a bit of information on timed attacks, we just need to know how to approach it, i believe we can get move times, when an enemy is using a move and precalculate how long we need to time before a timing attack or block, as it seems like the animation travel is static based on the code.
+- Spell Locks
+  - for the bomb characters in the demo, breaking a spell lock will kill it - need to find records for that.
+- Loot, did it drop and should we grab it?
+
+Combat Manager
+  - 0xF0 - currentEncounter (Encounter)
+    - 0x110 - encounterDone (bool)
+    - 0x168 - inGameHud (InGameHud)
+      - 0x60 - playerBattleUI (PlayerBattleUI)
+        - 0x50 - comboPointsPanel (ComboPointsPanel)
+          - 0x40 - comboPointerMeter (ComboPointsMeter)
+            - 0x3C - currentComboPoints (float) 0 - 3
+    - 0xE0 - controller
+      - 0x80 - battleUI
+        - 0x40 - playerPanelsList
+          - 0x10 - _items
+            - 0x18 - Count
+            - 0x20 - Item[0] - 0x08 width
+              - 0x28 - hpTextField
+                - 0x50 - currentvalue (the animations current value when losing health)
+                - 0x54 - targetValue (the final value, after animation)
+              - 0x30 - actionPointsTextField (MP)
+                - 0x50 - currentvalue (the animations current value when losing mp)
+                - 0x54 - targetValue (the final value, after animation)
+              - 0x70 - characterDefinitionId (do these vary between reloads?)
+              - 0x78 - selected (is active character for menus)
+
+    - 0x120 - playerActors (Generic.List<PlayerCombatActor>)
+      - 0x10 - _items
+        - 0x18 - count
+        - 0x20 - Item[0] (Player 0)
```

Comment:
Is the order always the same each combat, or do we have to track the Ids? Probably best to track Ids either way.

Reply:
We will have to track the IDs and probably the grid placement, because it doesn't even seem like its ordered at all.